autodocs.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+LINT=
+GENERATE=
+
+while getopts :lL:gG: opt
+do
+    case "${opt}" in
+        l)
+            LINT=1
+            ;;
+        g)
+            GENERATE=1
+            ;;
+        \?)
+            echo "ERROR: unknown parameter \"$OPTARG\""
+            exit 1
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+IFS=$'\n'
+# create an array of all unique directories containing .tf files 
+arr=($(find . -name '*.tf' | xargs -I % sh -c 'dirname %' | sort -u))
```

Comment:
should be able to combine the IFS with this line so it is set _only_ for this one line (no need to unset)

also, prefer var names that describe the value, not the type...

and no need to xargs for this one...

```suggestion
IFS=$'\n' directories=($(find . -name '*.tf' -exec dirname {} + | sort -u))
```

Reply:
I _think_ this is a macOS quirk but `$(find . -name '*.tf' -exec dirname {} +)` gives the super cryptic `usage: dirname path` error. Only way I was able to get this to work on my local machine was with `xargs`

Reply:
Try quoting the braces... `'{}'`?

Reply:
Ahhh, https://unix.stackexchange.com/a/503945

Try changing the `+` to `\;`

Reply:
that worked!